bin/p2-log-bridge/main.go

```diff
@@ -20,31 +22,56 @@ func main() {
 	kingpin.Version(version.VERSION)
 	kingpin.Parse()
 
+	if err := setSTDINToBlock(); err != nil {
+		logging.DefaultLogger.WithError(err).Error("fatal error setting STDIN to block")
+		os.Exit(1)
+	}
+
+	var wg sync.WaitGroup
 	loggerCmd := exec.Command(*durableLogger)
 	durablePipe, err := loggerCmd.StdinPipe()
 	if err != nil {
-		logging.DefaultLogger.WithError(err).Errorln("failed during logbridge setup")
-		os.Exit(1)
+		logging.DefaultLogger.WithError(err).Error("fatal error during configuration of subordinate log command")
 	}
 
-	go handleSignals(durablePipe)
+	wg.Add(1)
+	go func(loggerCmd exec.Cmd, wg sync.WaitGroup) {
+		defer wg.Done()
 
-	go func(r io.Reader, durableWriter io.Writer, nonDurableWriter io.Writer, logger logging.Logger) {
-		logbridge.Tee(r, durableWriter, nonDurableWriter, logger)
-	}(os.Stdin, durablePipe, os.Stdout, logging.DefaultLogger)
+		if err := loggerCmd.Start(); err != nil {
+			logging.DefaultLogger.WithError(err).Errorln("fatal error during execution of subordinate log command")
+			os.Exit(1)
+		}
 
-	loggerCmd.Start()
-	err = loggerCmd.Wait()
-	if err != nil {
-		logging.DefaultLogger.WithError(err)
-		os.Exit(1)
-	}
+		err := loggerCmd.Wait()
+		if err != nil {
+			logging.DefaultLogger.WithError(err).Errorln("fatal error in subordinate log command")
+			os.Exit(1)
+		}
+	}(*loggerCmd, wg)
+
+	logbridge.Tee(os.Stdin, durablePipe, os.Stdout, logging.DefaultLogger)
```

Comment:
Well, the way it's written this command could be wired up via shell pipe to a potentially slow log forwarder. 

I'm thinking of this script as an example of how to use the logbridge package rather than a reference implementation. I did find it a bit difficult to emulate our lossy samsa writer without pulling in a lot of dependencies. 

If you feel strongly, I could add a second argument to allow a lossy cmd to be specified on the command line here. That would definitely illustrate how we use logbridge internally more clearly at the cost of a bit more development time.